tests_e2e/tests/lib/network_security_rule.py

```diff
@@ -76,14 +76,14 @@ def _get_network_security_group(self) -> Dict[str, Any]:
                 "securityRules": []
             }}
         }}""")
-        resources.append(network_security_group)
+        nsg_reference = "network_security_groups"
+        resources[nsg_reference] = network_security_group
 
         #
         # Add a dependency on the NSG to the virtual network
         #
         network_resource = UpdateArmTemplate.get_resource(resources, "Microsoft.Network/virtualNetworks")
         network_resource_dependencies = network_resource.get("dependsOn")
-        nsg_reference = f"[resourceId('Microsoft.Network/networkSecurityGroups', '{self._NETWORK_SECURITY_GROUP}')]"
         if network_resource_dependencies is None:
             network_resource["dependsOn"] = [nsg_reference]
```

Comment:
now the virtual network has a dependency on "network_security_groups", is that correct? I think the dependency should still be nsg_reference = f"[resourceId('Microsoft.Network/networkSecurityGroups', '{self._NETWORK_SECURITY_GROUP}')]"

Reply:
no change in dependency, the reference we need to use will change instead resource id, it will be dict key where this resource id kept.

Reply:
for example, https://github.com/microsoft/lisa/pull/3163/files#diff-6ade80587709e19354bf9ad7752936686e13956b007be37baadc9baa14b2d733R699